crates/tap-agent/src/agent/sender_account.rs

```diff
@@ -603,10 +605,20 @@ impl Actor for SenderAccount {
             .with_label_values(&[&sender_id.to_string()])
             .set(config.trigger_value as f64);
 
-        let sender_aggregator = HttpClientBuilder::default()
-            .request_timeout(config.rav_request_timeout)
-            .build(&sender_aggregator_endpoint)?;
+        let endpoint = Endpoint::new(sender_aggregator_endpoint.to_string())
+            .context("Failed to create an endpoint for the sender aggregator")?;
 
+        let sender_aggregator = TapAggregatorClient::connect(endpoint.clone())
+            .await
+            .with_context(|| {
+                format!(
+                    "Failed to connect to the TapAggregator endpoint '{}'",
+                    endpoint.uri()
+                )
+            })?;
+        #[cfg(not(test))]
+        let sender_aggregator =
+            sender_aggregator.send_compressed(tonic::codec::CompressionEncoding::Zstd);
```

Comment:
Can you comment on this please? Quick note to reader about why this is needed for testing and not otherwise, please!

Reply:
Done!